### PR TITLE
raftstore: Add hibernate region metrics

### PR DESCRIPTION
**Pull request by yiwu-arbug**
### What problem does this PR solve?

Adding hibernate region metrics

Problem Summary:

### What is changed and how it works?

What's Changed:
Adding the following metrics
* region count in different hibernate state, breakdown by leader/non-leader
* leader count in different hibernate leader state (awaken/poll/hibernated)

example:
It is in the "Server" pannel.
Screen Shot 2021-05-26 at 11 27 55 PM

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

- No release notes

### CODE REVIEW

**Comment by yiwu-arbug** [comment]
Work in progress. Not tested yet.

File: metrics/grafana/tikv_details.json [hunk 1]
```diff
@@ -3540,6 +3540,212 @@
...
+              "legendFormat": "{{instance}}-{{role}}-{{state}}",
+              "refId": "A",
+              "step": 10
+            }
+          ],
+          "thresholds": [],
+          "timeFrom": null,
+          "timeRegions": [],
+          "timeShift": null,
+          "title": "Hibernate Regions",
```
Comment (BusyJay) [inline metrics/grafana/tikv_details.json:3609]: Should be "Hibernate Peers".

**Review by BusyJay — COMMENTED** [review]
How about collecting the state on raft base tick where almost all state transition take place? Using coprocessor seems too complicated than I thought comparing *2* metrics and *752* updates.

**Comment by yiwu-arbug** [comment]
> How about collecting the state on raft base tick where almost all state transition take place? Using coprocessor seems too complicated than I thought comparing _2_ metrics and _752_ updates.

I think basically we will need a place to hold the state for all peers, and a worker to flush the metrics. I look through the code and seems the region info accessor fits the needs. What do you think?

**Comment by yiwu-arbug** [comment]
Per @BusyJay 's request, remove region_info_accessor logic and change to use a simple gauge.

**Review by 5kbpers — APPROVED** [review]

**Review by NingLin-P — APPROVED** [review]
LGTM

**Pull request merged**